OpenDreamClient/Rendering/DreamIcon.cs

```diff
@@ -127,31 +151,225 @@ internal sealed class DreamIcon(IGameTiming gameTiming, ClientAppearanceSystem a
+                appearance.Layer = ((1-factor) * _appearance.Layer) + (factor * endAppearance.Layer);
+            }
+
+            /* TODO luminosity
+            if (endAppearance.Luminosity != _appearance.Luminosity) {
+                appearance.Luminosity = ((1-factor) * _appearance.Luminosity) + (factor * endAppearance.Luminosity);
+            }
+            */
+
+            /* TODO maptext
+            if (endAppearance.MapTextWidth != _appearance.MapTextWidth) {
+                appearance.MapTextWidth = (ushort)Math.Clamp(((1-factor) * _appearance.MapTextWidth) + (factor * endAppearance.MapTextWidth), 0, 65535);
+            }
+
+            if (endAppearance.MapTextHeight != _appearance.MapTextHeight) {
+                appearance.MapTextHeight = (ushort)Math.Clamp(((1-factor) * _appearance.MapTextHeight) + (factor * endAppearance.MapTextHeight), 0, 65535);
+            }
+
+            if (endAppearance.MapTextX != _appearance.MapTextX) {
+                appearance.MapTextX = (short)Math.Clamp(((1-factor) * _appearance.MapTextX) + (factor * endAppearance.MapTextX), -32768, 32767);
+            }
+
+            if (endAppearance.MapTextY != _appearance.MapTextY) {
+                appearance.MapTextY = (short)Math.Clamp(((1-factor) * _appearance.MapTextY) + (factor * endAppearance.MapTextY), -32768, 32767);
+            }
+            */
+
+            if (endAppearance.PixelOffset != _appearance.PixelOffset) {
+                Vector2 startingOffset = appearance.PixelOffset;
+                Vector2 newPixelOffset = Vector2.Lerp(startingOffset, endAppearance.PixelOffset, 1.0f-factor);
+
+                appearance.PixelOffset = (Vector2i)newPixelOffset;
+            }
+
+            if (!endAppearance.Transform.SequenceEqual(_appearance.Transform)) {
+                appearance.Transform[0] = (1.0f-factor)*_appearance.Transform[0] + (factor * endAppearance.Transform[0]);
+                appearance.Transform[1] = (1.0f-factor)*_appearance.Transform[1] + (factor * endAppearance.Transform[1]);
+                appearance.Transform[2] = (1.0f-factor)*_appearance.Transform[2] + (factor * endAppearance.Transform[2]);
+                appearance.Transform[3] = (1.0f-factor)*_appearance.Transform[3] + (factor * endAppearance.Transform[3]);
+                appearance.Transform[4] = (1.0f-factor)*_appearance.Transform[4] + (factor * endAppearance.Transform[4]);
+                appearance.Transform[5] = (1.0f-factor)*_appearance.Transform[5] + (factor * endAppearance.Transform[5]);
+            }
+
+            if (timeFactor >= 1f) {
+                if (animation.loops > 0)
+                {
+                    var tempAnimation = _appearanceAnimations[i];
+                    tempAnimation.loops--;
+                    _appearanceAnimations[i] = tempAnimation;
+                }
+                if (animation.loops == 0) {
+                    toRemove ??= new();
+                    toRemove!.Add(animation);
```

Comment:
C# can already tell it's not null if `??= new()` is done beforehand.
```suggestion
                    toRemove.Add(animation);
```